packages/aws-rfdk/lib/deadline/lib/configure-spot-event-plugin.ts

```diff
@@ -0,0 +1,491 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as path from 'path';
+
+import {
+  IVpc,
+  SubnetSelection,
+  SubnetType,
+} from '@aws-cdk/aws-ec2';
+import {
+  Role,
+  Policy,
+  PolicyStatement,
+} from '@aws-cdk/aws-iam';
+import {
+  Code,
+  Function as LambdaFunction,
+  LayerVersion,
+  Runtime,
+} from '@aws-cdk/aws-lambda';
+import { RetentionDays } from '@aws-cdk/aws-logs';
+import { ISecret } from '@aws-cdk/aws-secretsmanager';
+import {
+  Construct,
+  CustomResource,
+  Duration,
+  Stack,
+} from '@aws-cdk/core';
+import { ARNS } from '../../lambdas/lambdaLayerVersionArns';
+import {
+  InternalSpotEventPluginSettings,
+  SEPConfiguratorResourceProps,
+} from '../../lambdas/nodejs/configure-spot-event-plugin';
+import { IRenderQueue, RenderQueue } from './render-queue';
+import { SpotEventPluginFleet } from './spot-event-plugin-fleet';
+import { SpotFleetRequestConfiguration} from './spot-event-plugin-fleet-ref';
+import { Version } from './version';
+import { IVersion } from './version-ref';
+
+/**
+ * How the event plug-in should respond to events.
+ */
+export enum SpotEventPluginState {
+  /**
+   * The Render Queue, all jobs and Workers will trigger the events for this plugin.
+   */
+  GLOBAL_ENABLED = 'Global Enabled',
+
+  /**
+   * No events are triggered for the plugin.
+   */
+  DISABLED = 'Disabled',
+}
+
+/**
+ * Logging verbosity levels for the Spot Event Plugin.
+ */
+export enum SpotEventPluginLoggingLevel {
+  /**
+   * Standard logging level.
+   */
+  STANDARD = 'Standard',
+
+  /**
+   * Detailed logging about the inner workings of the Spot Event Plugin.
+   */
+  VERBOSE = 'Verbose',
+
+  /**
+   * All Verbose logs plus additional information on AWS API calls that are used.
+   */
+  DEBUG = 'Debug',
+
+  /**
+   * No logging enabled.
+   */
+  OFF = 'Off',
+}
+
+/**
+ * How the Spot Event Plugin should handle Pre Job Tasks.
+ * See https://docs.thinkboxsoftware.com/products/deadline/10.1/1_User%20Manual/manual/job-scripts.html
+ */
+export enum SpotEventPluginPreJobTaskMode {
+  /**
+   * Only start 1 Spot instance for the pre job task and ignore any other tasks for that job until the pre job task is completed.
+   */
+  CONSERVATIVE = 'Conservative',
+
+  /**
+   * Do not take the pre job task into account when calculating target capacity.
+   */
+  IGNORE = 'Ignore',
+
+  /**
+   * Treat the pre job task like a regular job queued task.
+   */
+  NORMAL = 'Normal',
+}
+
+/**
+ * The Worker Extra Info column to be used to display AWS Instance Status
+ * if the instance has been marked to be stopped or terminated by EC2 or Spot Event Plugin.
+ * See "AWS Instance Status" option at https://docs.thinkboxsoftware.com/products/deadline/10.1/1_User%20Manual/manual/event-spot.html#event-plugin-configuration-options
+ * and https://docs.thinkboxsoftware.com/products/deadline/10.1/1_User%20Manual/manual/worker-config.html#extra-info
+ */
+export enum SpotEventPluginAwsInstanceStatus {
```

Comment:
What if this were renamed to be `SpotEventPluginDisplayInstanceStatus`? It seems like a more accurate typename, plus it makes sense when reading code that uses it.

Reply:
Sure, renamed.